application-task-default/src/main/java/com/xwiki/task/internal/notifications/TaskChangedEventNotificationListener.java

```diff
@@ -0,0 +1,285 @@
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.Date;
+
+import javax.inject.Inject;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.suigeneris.jrcs.rcs.Version;
+// import org.xwiki.bridge.DocumentAccessBridge;
+import org.xwiki.bridge.event.DocumentCreatedEvent;
+import org.xwiki.bridge.event.DocumentUpdatedEvent;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.manager.ComponentLookupException;
+import org.xwiki.component.manager.ComponentManager;
+// import org.xwiki.model.reference.DocumentReference;
+// import org.xwiki.model.reference.EntityReference;
+import org.xwiki.observation.AbstractEventListener;
+import org.xwiki.observation.ObservationManager;
+import org.xwiki.observation.event.Event;
+// import org.xwiki.user.CurrentUserReference;
+// import org.xwiki.user.UserReference;
+// import org.xwiki.notifications.NotificationException;
+// import org.xwiki.notifications.filters.watch.WatchedEntitiesManager;
+// import org.xwiki.notifications.filters.watch.WatchedEntityFactory;
+// import org.xwiki.notifications.filters.watch.WatchedEntityReference;
+// import org.xwiki.user.UserReferenceResolver;
+// import org.xwiki.user.internal.document.DocumentUserReference;
+
+import com.xpn.xwiki.doc.DocumentRevisionProvider;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.doc.XWikiDocumentArchive;
+import com.xpn.xwiki.objects.BaseObject;
+import com.xpn.xwiki.objects.StringProperty;
+// import com.xpn.xwiki.user.api.XWikiUser;
+import com.xwiki.task.model.Task;
+import com.xpn.xwiki.objects.LargeStringProperty;
+import com.xpn.xwiki.objects.PropertyInterface;
+import com.xpn.xwiki.objects.DateProperty;
+
+/**
+ * Listener which fires when modifying a task in order to notify users of the changes.
+ * @version $Id$
+ * @since 3.5.2
+ */
+@Component
+@Singleton
+@Named("com.xwiki.task.internal.notifications.TaskChangedEventNotificationListener")
+public class TaskChangedEventNotificationListener extends AbstractEventListener
+{
+    private static final String TASK_MANAGER_CLASS_NAME = "TaskManager.TaskManagerClass";
+
+    /**
+     * The fields of the Task class which are watched for changes.
+     */
+    private static final List<String> WATCHED_FIELDS = Arrays.asList(
+        Task.ASSIGNEE,
+        Task.DUE_DATE,
+        Task.PROJECT,
+        Task.STATUS,
+        Task.SEVERITY
+    );
+
+    @Inject
+    private ComponentManager componentManager;
+
+    @Inject
+    private DocumentRevisionProvider documentRevisionProvider;
+
+    @Inject
+    private Logger logger;
+
+    private ObservationManager observationManager;
+
+    /**
+     * Initialize the listener.
+     */
+    public TaskChangedEventNotificationListener()
+    {
+        super(TaskChangedEventNotificationListener.class.getName(), Arrays.asList(new DocumentUpdatedEvent(), 
```

Comment:
Would be best to listen to `XObjectPropertyUpdatedEvent` and / or `XObjectPropertyAddedEvent`